src/3d/qgsvectorlayerchunkloader_p.cpp

```diff
@@ -203,4 +205,63 @@ void QgsVectorLayerChunkedEntity::onTerrainElevationOffsetChanged( float newOffs
   mTransform->setTranslation( QVector3D( 0.0f, newOffset, 0.0f ) );
 }
 
+QVector<QgsRayCastingUtils::RayHit> QgsVectorLayerChunkedEntity::rayIntersection( const QgsRayCastingUtils::Ray3D &ray, const QgsRayCastingUtils::RayCastContext &context ) const
+{
+  return QgsVectorLayerChunkedEntity::rayIntersection( activeNodes(), mTransform->matrix(), ray, context );
+}
+
+QVector<QgsRayCastingUtils::RayHit> QgsVectorLayerChunkedEntity::rayIntersection( const QList<QgsChunkNode *> &activeNodes, const QMatrix4x4 &transformMatrix, const QgsRayCastingUtils::Ray3D &ray, const QgsRayCastingUtils::RayCastContext &context )
+{
+  Q_UNUSED( context )
```

Comment:
I'd rather keep this in this file, as all chunked entities' rayIntersection implementations are in their respective files. The static method exists only so that it can be re-used by the rule-based code.
I made this private and friended QgsRuleBasedChunkedEntity so it can access it instead.
Ideally, I'd like to completely remove QgsRuleBasedChunkedEntity at some point, as conceptually it is not any different to QgsVectorLayerChunkedEntity

Reply:
may be you could make `QgsRuleBasedChunkedEntity` inherit from `QgsVectorLayerChunkedEntity`? This should resolve both problems.

Reply:
I thought about that too, but conceptually those two do not have a parent-child relation, they are rather siblings if not the same thing. I suggest we discuss this when we meet at the hackfest!